src/TLuaInterpreter.cpp

```diff
@@ -14281,7 +14281,22 @@ void TLuaInterpreter::parseJSON(QString& key, const QString& string_data, const
         if (match.capturedStart() != -1) {
             QString title = match.captured(1);
             QString initialText = match.captured(2);
-            initialText.replace(QStringLiteral(R"(\n)"), QStringLiteral("\n"));
+            int j=0;
+            while((j = initialText.indexOf('\\', j)) != -1){
+                // while we find a backslash, look at the next character, decide what to replace them both with
+                switch(initialText.at(j+1).unicode()){
```

Comment:
Yes, `j` can indeed cause it to go beyond the string.  The `at(index)` returns zero if the character index that you're looking for is beyond the length of string.  For example, if the game gave you `abc\` for you to try and unescape, then it will find the backslash and then look ahead but get a zero.  When that gets fed into the switch then there isn't a case for null character and so it just skips it and is finished.

Similar with input of `\u`.  It'll find the backslash and the `u`, then looks at the next four bytes. 
 Even the first of those is out of range, but midRef(index,size) is alright with that.  Looking back at that part now though, I think tomorrow I'm going to just make it handle any number, no reason not to.

Reply:
> ...  The at(index) returns zero if the character index that you're looking for is beyond the length of string ...

I think you are wrong there, https://doc.qt.io/qt-5/qstring.html#at says:
> `const QChar QString::at(int position) const`
>
> Returns the character at the given index *position* in the string.
>
> The *position* must be a valid index position in the string (i.e., 0 <= *position* < `size()`).
>
> **See also** `operator[]()`.

IIRC using an out of bounds value for *position* (`index`) will not return a null `QString` but instead will cause the feared *undefined behaviour* ... :fearful:

Reply:
You're right, I think I was looking at something outdated.

https://doc.qt.io/archives/2.3/qstring.html#4f6a13

`QChar QString::at ( uint ) const`
`Returns the character at i, or 0 if i is beyond the length of the string.`